initramfs/src/registry.rs

```diff
@@ -50,48 +154,180 @@ impl Registry {
             .map_err(|e| anyhow!(e).context("Failed to parse response"))?
             .token;
 
-        debug!("Successfully got auth token : {:?}", token);
+        debug!("Successfully got auth token");
+        trace!("Token: {}", token);
 
         Ok(token)
```

Comment:
Instead of returning the token here, you could just set self.token

Another option, as you suggested, would be to make get_token() a static function

Reply:
Could you elaborate on why we would want to move this to either static (which is not convenient since we need `self.auth_url`) or set it directly within the method ?

Reply:
The point about setting it directly was to prevent the token from being moved around in an unwanted way. 

The idea a the static function was that having static functions is easier to test and debug, and since it can be achieved using only one more parameter to the function it was more of a "why not do this" on my side. 

Obviously both of this option can't be done at the same time, if I had to choose i would prefer the self.set_token() option

Reply:
Sounds good to me, will do.